mempool/v1/mempool.go

```diff
@@ -469,7 +481,10 @@ func (txmp *TxMempool) addNewTransaction(wtx *WrappedTx, checkTxRes *abci.Respon
 			"post_check_err", err,
 		)
 
-		txmp.metrics.FailedTxs.Add(1)
+		txmp.metrics.FailedTxs.With(mempool.FailedAdding).Add(1)
+		if err != nil {
```

Comment:
This should also trigger of abci code is not zero. In this case we should print the raw logs and code as the reason

Reply:
https://github.com/celestiaorg/celestia-core/pull/1158/commits/190b73169041652a0341215f2d051b1757bc0ef3